src/Microsoft.Health.Dicom.Functions/DataCleanup/DataCleanupDurableFunction.Orchestration.cs

```diff
@@ -0,0 +1,90 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Functions.DataCleanup.Models;
+using Microsoft.Health.Operations.Functions.DurableTask;
+
+namespace Microsoft.Health.Dicom.Functions.DataCleanup;
+
+public partial class DataCleanupDurableFunction
+{
+    /// <summary>
+    /// Asynchronously cleans up instance data.
+    /// </summary>
+    /// <remarks>
+    /// Durable functions are reliable, and their implementations will be executed repeatedly over the lifetime of
+    /// a single instance.
+    /// </remarks>
+    /// <param name="context">The context for the orchestration instance.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>A task representing the <see cref="DataCleanupAsync"/> operation.</returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="context"/> or <paramref name="logger"/> is <see langword="null"/>.
+    /// </exception>
+    /// <exception cref="FormatException">Orchestration instance ID is invalid.</exception>
+    [FunctionName(nameof(DataCleanupAsync))]
+    public async Task DataCleanupAsync(
+        [OrchestrationTrigger] IDurableOrchestrationContext context,
+        ILogger logger)
+    {
+        EnsureArg.IsNotNull(context, nameof(context)).ThrowIfInvalidOperationId();
+        logger = context.CreateReplaySafeLogger(EnsureArg.IsNotNull(logger, nameof(logger)));
+
+        DataCleanupCheckPoint input = context.GetInput<DataCleanupCheckPoint>();
+
+        IReadOnlyList<WatermarkRange> batches = await context.CallActivityWithRetryAsync<IReadOnlyList<WatermarkRange>>(
```

Comment:
If we make this more generic, perhaps we can make an extension method for this call

Reply:
Sure, will keep in my next PR